db/migrate/20250814153656_add_compound_primary_key_to_mhv_metrics_unique_user_events.rb

```diff
@@ -0,0 +1,14 @@
+class AddCompoundPrimaryKeyToMHVMetricsUniqueUserEvents < ActiveRecord::Migration[7.2]
+  disable_ddl_transaction!
+
+  def change
+    # Add unique compound index - columns already have NOT NULL constraints from table creation
+    # This ensures one record per user per event type
+    add_index :mhv_metrics_unique_user_events, 
+              [:user_id, :event_name], 
+              unique: true,
+              algorithm: :concurrently,
+              if_not_exists: true,
+              name: 'index_mhv_metrics_unique_user_events_on_user_id_and_event_name'
+  end
+end
```

Comment:
When creating database migrations, it is good practice to run the migrate operation locally, then run the rollback operation to verify the migration can be undone. e.g. - `bundle exec rails db:rollback STEP=2` to rollback the last two migrations.